### DISCUSSION TITLE
Adding additional Coins

### DISCUSSION THREAD

**Discussion by duderino999** [hulatang/ceres-combineharvester#19] [Ideas]
Hi,

first of all, great work so far. Using it actually for 32 forks and it's working really stable. There are a few coins "missing", would be great, if they could be added as well.

Chaingreen (unofficial update): https://github.com/morrillup/chaingreen-blockchain
Kujenga: https://github.com/Kujenga-Network/kujenga-blockchain
Lucky: https://github.com/dream80/lucky-blockchain
mELON: https://github.com/mELON-Network/melon-blockchain

Cheers,
duderino999

**Comment by duderino999** [comment]
I saw that the configs for Chaingreen and Lucky are already existing, but they have not been included within coins_config.yaml, if I add them manually, it looks like they are working.

**Reply by hulatang** [reply]
Thank your for your feedback and references.
coins you mentioned will be tested and ceres will update as soon as they are available.

Chaingreen and lucky are not yet tested, and attributes may be missing

**Reply by duderino999** [reply]
Great and thanks for taking care of it. There is one additional, that is actually not included and might be worth to implement:

Xcha: https://github.com/xchaio/xcha-blockchain

**Reply by hulatang** [reply]
> Great and thanks for taking care of it. There is one additional, that is actually not included and might be worth to implement:
> 
> Xcha: https://github.com/xchaio/xcha-blockchain

Thanks for your feedback,  and I will go to check if it is possible to include xcha

**Reply by hulatang** [reply]
> Great and thanks for taking care of it. There is one additional, that is actually not included and might be worth to implement:
> 
> Xcha: https://github.com/xchaio/xcha-blockchain

unfortunately, after reviewing xcha's code, currently it can not be supported by Ceres, cause they change Chia's message protocol between farmer and harvester.

**Reply by duderino999** [reply]
ok, thanks for looking at it.

**Comment by duderino999** [comment]
Quick question, what are the requirements to add new coins? I tried to add a new one myself, I modified/added following files:

- /ceres/consensus/all_coins_default_constants/xyz_new_coin_default_constants.py (modified this file with the values from the default_constants.py of the fork coin itself)

- /ceres/util/coins_initial_config/xyz_new_coin_initial_config.yaml (modified this file with the values from the initial_config.yaml of the fork coin itself)

- added the fork coin under ceres-initial-coins-config.yaml

Then followed the standard setup with init, init --coins, copy ca and generate_ssl.

After starting ceres, its's giving me a handshake error between ceres and the farmer of the coin. So I am assuming something is missing here or the coin is not compatible with ceres.

**Reply by duderino999** [reply]
debugging the farmer of the fork coin provides me following message:

2021-11-12T13:09:50.477 farmer farmer_server              : DEBUG    Invalid connection type for connection 127.0.0.1, while closing. Handshake never finished.
2021-11-12T13:09:50.477 farmer farmer_server              : WARNING  Incompatible network ID. Maybe the peer is on another network

i was assuming this is related to a wrong mainnet/testnet setup, but when double checking ceres and fork coin, both are set to mainnet.

**Reply by hulatang** [reply]
pls update to the newest version which fix a bug related to this
thank you.

**Reply by duderino999** [reply]
yes, did it yesterday ... it's working now as expected. I have also added Kujenga and Shibgreen locally. They are working also as expected. Would be great to add them as well. I have attached the files for your reference.
[kujenga_shibgreen.zip](https://github.com/hulatang/ceres-combineharvester/files/7548276/kujenga_shibgreen.zip)
.

**Comment by duderino999** [comment]
Quick question on Chaingreen and Lucky. Any chance they will be included within Ceres?

**Reply by hulatang** [reply]
changreen and lucky can not be supported by ceres like normal, cause they change Chia harvester_farmer  message protocol.

in order to make these kind of forks work with ceres, a patch file is needed, 
If people would like to override a file under their farmer codes with this patch file, this will be an option

**Reply by duderino999** [reply]
i can give it a try. can you highlight what exactly needs to be patched?

**Reply by hulatang** [reply]
some patch files will be supplied for testing

**Reply by duderino999** [reply]
i just wanted to check, if you had already some time working on the patch files?

**Comment by duderino999** [comment]
Another one. As you might know, Silicoin has deployed their release with staking and they changed for that the consensus. With the staking functionality the actual Silicoin release is not working anymore with Ceres. Are you planning to fix that, if possible? Or can we assume, that it won't work anymore with Ceres due to their staking functionality. Their latest release is open source again, so might be worth to have a look at it.

**Reply by hulatang** [reply]
silicoin's code has been reviewed very carefully to check if it is possible to be supported by ceres.
unfortunately ceres can not support silicoin in a standard way.
It will require to modify the silicoin farmer's code by yourself,
I'm not sure if it is worth to do in this way

**Reply by duderino999** [reply]
it depends on, how much modification needs to be done. always happy to give it a try.

**Reply by hulatang** [reply]
fine, once is tested I will let you know.
thanks

**Comment by duderino999** [comment]
I have added Goji to Ceres, as it was updated recently. Find attached the relevant files, in case you want to update Ceres.
[goji.zip](https://github.com/hulatang/ceres-combineharvester/files/7739050/goji.zip)

**Comment by duderino999** [comment]
Hi,

are you still working on this project? Curious to hear, as there hasn't been any updates for a few months. If you are still working on it, there was a hard fork of LittleLamboCoin and the old chain can be threated as abandoned. Would be great, if you could have a look at the new chain and if it can be added to Ceres:

https://github.com/BTCgreen-Network/littlelambocoin-blockchain/

Cheers
